### ISSUE TITLE
Globally installed 7.0-alpha breaks restore for 6.0 projects

### ISSUE THREAD

**Issue by ManickaP** [dotnet/sdk#22752]
After global installation of `7.0.100-alpha.1.21568.2` SDK via `dotnet-install.sh`, restore for projects targeting `net6.0` git broken with this error:
```
  Determining projects to restore...
/home/manicka/work/boo/boo.csproj : error NU1102: Unable to find package Microsoft.NETCore.App.Ref with version (= 6.0.0-rc.2.21452.2)
/home/manicka/work/boo/boo.csproj : error NU1102:   - Found 35 version(s) in nuget.org [ Nearest version: 6.0.0-rc.2.21480.5 ]
/home/manicka/work/boo/boo.csproj : error NU1102: Unable to find package Microsoft.NETCore.App.Host.linux-x64 with version (= 6.0.0-rc.2.21452.2)
/home/manicka/work/boo/boo.csproj : error NU1102:   - Found 68 version(s) in nuget.org [ Nearest version: 6.0.0-rc.2.21480.5 ]
/home/manicka/work/boo/boo.csproj : error NU1102: Unable to find package Microsoft.AspNetCore.App.Ref with version (= 6.0.0-rc.2.21452.1)
/home/manicka/work/boo/boo.csproj : error NU1102:   - Found 39 version(s) in nuget.org [ Nearest version: 6.0.0-rc.2.21480.10 ]
```

Output of `dotnet --info`:
```
.NET SDK (reflecting any global.json):
 Version:   7.0.100-alpha.1.21568.2
 Commit:    0be3213624

Runtime Environment:
 OS Name:     arch
 OS Version:
 OS Platform: Linux
 RID:         arch-x64
 Base Path:   /usr/share/dotnet/sdk/7.0.100-alpha.1.21568.2/

Host (useful for support):
  Version: 7.0.0-alpha.1.21567.1
  Commit:  10e107deba

.NET SDKs installed:
  6.0.100 [/usr/share/dotnet/sdk]
  7.0.100-alpha.1.21568.2 [/usr/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 7.0.0-alpha.1.21567.15 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.0 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.0-alpha.1.21567.1 [/usr/share/dotnet/shared/Microsoft.NETCore.App]

To install additional .NET runtimes or SDKs:
  https://aka.ms/dotnet-download
```

Tested on 3 Linux machines: 2x Arch linux, 1x Pop!_OS (21.04), all manifest the same error.

EDIT: After manual removal of 7.0, everything works as expected.

**Comment by kirsan31** [comment]
.Net6 build is [broken too](https://github.com/dotnet/winforms/issues/3202#issuecomment-978891569).

After installing .Net7 sdk (`7.0.100-alpha.1.21568.2`) and enabling this in VS2022:
![image](https://user-images.githubusercontent.com/17767561/143394002-dbb03e05-4673-4ff3-b24f-76e3db7e35b2.png)

`net6.0-windows` targets will fail to build with dozen's of import errors:

![image](https://user-images.githubusercontent.com/17767561/143394579-f2014fdd-190d-4b36-abf1-1fd6bb960ed3.png)
![image](https://user-images.githubusercontent.com/17767561/143394586-69646ded-429c-478b-a4fb-253128f0c7af.png)

These targets, still builds fine: `net7.0-windows;net5.0-windows;netcoreapp3.1;net472`...

**Comment by sfoslund** [comment]
It sounds like you're missing some feeds. Can you try following the instructions here:https://github.com/dotnet/installer#installers-and-binaries to add the missing feeds?

**Comment by ManickaP** [comment]
Adding 6.0 feed `<add key="dotnet6" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet6/nuget/v3/index.json" />` to nuget.config helped. Thanks!

**Comment by AraHaan** [comment]
The odd issue is that really it should be using the stable .NET 6 apphost and runtimes when in the main branch.

**Comment by elinor-fung** [comment]
For anyone else that found this issue and was confused by why explicitly adding the feed would be necessary for a released 6.0 build, the PR https://github.com/dotnet/installer/pull/12654 should fix this (issue in installer repo: https://github.com/dotnet/installer/issues/12968).
